src/x86/runtime.rs

```diff
@@ -211,7 +211,52 @@ fn detect_features() -> usize {
         value = set_bit(value, __Feature::sse4_2 as u32);
     }
     if test_bit(proc_info_ecx, 21) {
-        value = set_bit(value, __Feature::tbm as u32);
+        // The TBM bit is only meaningfully set in AMD CPUs:
+        let ebx: u32;
+        let edx: u32;
+        let ecx: u32;
+        // EAX = 0: Basic Information. The vendor ID is stored in 12 u8 ascii
+        // chars, returned in EBX, EDX, and ECX (in that order):
+        unsafe {
+            asm!("cpuid"
+                 : "={ebx}"(ebx), "={ecx}"(ecx), "=(edx)"(edx)
+                 : "{eax}"(0x0000_0000_u32), "{ecx}"(0 as u32)
+                 : :);
+        }
+        let ebx: [u8; 4] = unsafe { ::std::mem::transmute(ebx) };
+        let edx: [u8; 4] = unsafe { ::std::mem::transmute(edx) };
+        let ecx: [u8; 4] = unsafe { ::std::mem::transmute(ecx) };
+        let vendor_id = [
+            ebx[0],
+            ebx[1],
+            ebx[2],
+            ebx[3],
+            ecx[0],
+            ecx[1],
+            ecx[2],
+            ecx[3],
+            edx[0],
+            edx[1],
+            edx[2],
+            edx[3],
+        ];
+        let vendor_id_amd: [u8; 12] = [
```

Comment:
I think you could use a byte literal for this? e.g., `b"AuthenticAMD"`. Unlike string literals, the type of a byte literal is a fixed sized `u8` array.

Reply:
Indeed, fixed :)